### PR TITLE
Generate JSON schemas from SQLAlchemy models

### PR DESCRIPTION
**Pull request by katstevens**
Trello: https://trello.com/c/6ptnNYg3/8-start-mapping-g-cloud-model-schemas-for-the-current-api

Trying out the `alchemyjsonschema` library for generating JSON schemas for our DB models. 

The aim is to get a representation of our database tables and fields, their types/requirements, and their relationships to other tables. 

Some tweaks were required to get the library to work with our DB, as the library can't parse `postgresql.JSON` field types (e.g. `Brief.data`) or fields that don't match the column name (e.g. `Brief._lot_id`).

**Note for reviewers**: this may look huge but there are only 2 Python files (`generate_model_schemas.py` and `test_generate_model_schemas.py`).

Out of scope: 
- Validating a model instance against its schema
- Schemas for `modelname.serialize()` 
- Schemas for API request/responses
- Automatically generating relationships/linked objects (limitation of the library)
- Automatically generating real paths to sub schemas for JSON-type fields e.g. `BriefResponse.award_details`. This is to avoid renaming schemas used elsewhere in the code.

### CODE REVIEW

File: app/generate_model_schemas.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    FrameworkLot, Lot, Framework, ContactInformation, Supplier, SupplierFramework,
+    FrameworkAgreement, User, Service, ArchivedService, DraftService, AuditEvent, Brief, BriefUser,
+    BriefResponse, BriefClarificationQuestion, DirectAwardProject, DirectAwardProjectUser, DirectAwardSearch,
+    DirectAwardSearchResultEntry, Outcome
+)
+
+from alchemyjsonschema import SchemaFactory, ForeignKeyWalker
+
+
+class DMForeignKeyWalker(ForeignKeyWalker):
```
Comment (katstevens) [inline app/generate_model_schemas.py]: Subclassed from https://github.com/podhmo/alchemyjsonschema/blob/master/alchemyjsonschema/__init__.py#L155

**Comment by lfdebrux** [comment]
Could we add `alchemyjsonschema` to requirements-dev instead of requirements-app?

**Comment by lfdebrux** [comment]
Code looks good to me

**Comment by lfdebrux** [comment]
I'm just worried about adding unnecessary dependencies to the docker image.

Is it worth adding a line to the README explaining how the schema was generated for future reference?

**Comment by katstevens** [comment]
Good call on the README. I've mentioned there that `alchemyjsonschema` is now in dev requirements 😸

**Pull request merged**